File: kernel/core/agent_manager.go
package core

import "fmt"

type Agent interface {
	Start()
	Stop()
	Name() string
}

type AgentManager struct {
	agents map[string]Agent
}

func NewAgentManager() *AgentManager {
	return &AgentManager{
		agents: make(map[string]Agent),
	}
}

func (am *AgentManager) Register(name string, agent Agent) {
	am.agents[name] = agent
	fmt.Println("✅ Registered Agent:", name)
}

func (am *AgentManager) StartAll() {
	for _, agent := range am.agents {
		agent.Start()
	}
}

func (am *AgentManager) StopAll() {
	for _, agent := range am.agents {
		agent.Stop()
	}
}

File: orchestrator/src/core/agent_manager.ts
import { EventBus } from "./event_bus";
import { Logger } from "../utils/logger";

export interface OrchestratorAgent {
  name(): string;
  start(): void;
}

export class AgentManager {
  private agents: Map<string, OrchestratorAgent>;
  private eventBus: EventBus;
  private logger: Logger;

  constructor(eventBus: EventBus, logger: Logger) {
    this.agents = new Map();
    this.eventBus = eventBus;
    this.logger = logger;
  }

  register(agent: OrchestratorAgent) {
    this.agents.set(agent.name(), agent);
    this.logger.info("AGENT_MANAGER", `Registered agent: ${agent.name()}`);
  }

  start() {
    this.logger.info(
      "AGENT_MANAGER",
      `Starting ${this.agents.size} orchestrator agents`
    );
    for (const agent of this.agents.values()) {
      agent.start();
    }
  }
}
